Sources/wkhtmltopdf/Page.swift

```diff
@@ -1,18 +1,46 @@
 import Foundation
-import NIO
 
 public struct Page {
     let content: Data
+    let options: [Option]
 
-    public init(_ content: Data) {
+    public init(_ content: Data, options: [Option] = []) {
         self.content = content
+        self.options = options
     }
 
-    public init(_ content: ByteBuffer) {
-        self.content = Data(content.readableBytesView)
+    public init(_ content: String, options: [Option] = []) {
+        self.content = Data(content.utf8)
+        self.options = options
     }
 
-    public init(_ content: String) {
-        self.content = Data(content.utf8)
+    public init(url: String, options: [Option] = []) {
+        self.content = Data()
+        self.options = [.rawArg(url)] + options
+    }
+
+    /// If true, this page contains a URL rather than content to be written to a file
+    var isUrl: Bool {
+        content.isEmpty
+            && options.contains {
+                if case .rawArg(_) = $0 { return true }
+                return false
+            }
+    }
+
+    /// Get the URL if this page is a URL page
+    var url: String? {
+        guard isUrl else { return nil }
+        for option in options {
+            if case .rawArg(let url) = option {
+                return url
+            }
+        }
+        return nil
+    }
```

Comment:
```suggestion
    private var firstRawArgOption: String? {
        for  option in options {
            if case .rawArg(let arg) = option {
                return arg
            }
        }
        return nil
    }
    
    var isUrl: Bool {
        content.isEmpty && self.firstRawArgOption != nil
    }

    /// Get the URL if this page is a URL page
    var url: String? {
        content.isEmpty ? self.firstRawArgOption : nil
    }
```
You know, so the `url` accessor doesn't iterate the options twice to do the same thing.